infra/vsts_agent_linux_startup.sh

```diff
@@ -95,6 +96,94 @@ useradd \
 #add docker group to user
 usermod -aG docker vsts
 
+# The Bazel cache fills up the hard drive in less than a day, and is a real
+# pain to clean up, as it consists of a very large amount of read-only files in
+# a fairly large tree of read-only directories. Therefore, it looks like the
+# best way to manage it is to mount it on a partition and reset the partition.
+
+cat <<REMOUNT_CACHE > /root/remount_cache.c
+#include <stdlib.h>
+#include <stdio.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+int exists(const char* path) {
+	struct stat s;
+	return stat(path, &s) == 0 && S_ISDIR(s.st_mode);
+}
+
+int execute_f(char** args) {
+	int status;
+	char** child_env = { NULL };
+	pid_t pid = fork();
+	if (0 == pid) {
+		// child process
+		if (-1 == execve(args[0], args, child_env)) {
+			printf("Failed to start child process calling %s.", args[0]);
+			perror(0);
+			exit(1);
+		}
+		// unreachable: if execve succeeded, we're off to another executable
+	}
+	// parent
+	if (-1 == pid) {
+		printf("Failed to fork.");
+		perror(0);
+		exit(1);
+	}
+	if (-1 == waitpid(pid, &status, 0)) {
+		printf("Failed to wait for child process '%s'.", args[0]);
+		exit(1);
+	}
+	if (1 != WIFEXITED(status)) {
+		printf("Child not terminated: '%s', aborting.", args[0]);
+		exit(1);
+	}
+	if (0 != WEXITSTATUS(status)) {
+		printf("Child '%s' exited with status %d", args[0], WEXITSTATUS(status));
+		exit(WEXITSTATUS(status));
+	}
+	return 0;
+}
+
+#define execute(...) execute_f((char*[]) { __VA_ARGS__, NULL })
+
+int main() {
+	char* mount_point = "/home/vsts/.cache/bazel";
+	char* lost = "/home/vsts/.cache/bazel/lost+found";
+	char* file = "/root/cache_file";
```

Comment:
```suggestion
	const char* mount_point = "/home/vsts/.cache/bazel";
	const char* lost = "/home/vsts/.cache/bazel/lost+found";
	const char* file = "/root/cache_file";
```
while const in C is not that great, it’s still better than no const imho.